src/EventManagers/DefaultEventManager.php

```diff
@@ -22,25 +22,25 @@ public function before(JobProcessing $event): void
 
     public function after(JobProcessed $event): void
     {
-        $this->getUpdater()->update($event, [
-            'status' => $this->getEntity()::STATUS_FINISHED,
-            'finished_at' => Carbon::now(),
-        ]);
+        if (!$event->job->hasFailed()) {
+            $this->getUpdater()->update($event, [
+                'status' => $this->getEntity()::STATUS_FINISHED,
+                'finished_at' => Carbon::now(),
+            ]);
+        }
     }
 
     public function failing(JobFailed $event): void
     {
         $this->getUpdater()->update($event, [
-            'status' => $this->getEntity()::STATUS_FAILED,
+            'status' => ($event->job->attempts() !== $event->job->maxTries()) ? $this->getEntity()::STATUS_FAILED : $this->getEntity()::STATUS_RETRYING,
```

Comment:
I think this condition is wrong. It should be:
```php
($event->job->attempts() >= $event->job->maxTries()) ? $this->getEntity()::STATUS_FAILED : $this->getEntity()::STATUS_RETRYING
```

Reply:
@AidasK comment makes sense